rest/taskana-rest-spring/src/main/java/pro/taskana/rest/TaskCommentController.java

```diff
@@ -0,0 +1,190 @@
+      throws NotAuthorizedException, TaskNotFoundException {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Entry to getTaskComments(taskId= {})", taskId);
+    }
+
+    List<TaskComment> taskComments = taskService.getTaskComments(taskId);
+
+    TaskCommentListResource taskCommentListResource =
+        taskCommentResourceAssembler.toListResource(taskComments);
+
+    ResponseEntity<TaskCommentListResource> response = ResponseEntity.ok(taskCommentListResource);
+
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Exit from getTaskComments(), returning {}", response);
+    }
+
+    return response;
+  }
+
+  @DeleteMapping(path = Mapping.URL_TASK_COMMENT)
+  @Transactional(readOnly = true, rollbackFor = Exception.class)
+  public ResponseEntity<TaskCommentResource> deleteTaskComment(
+      @PathVariable String taskId, @PathVariable String taskCommentId)
+      throws NotAuthorizedException, TaskNotFoundException, TaskCommentNotFoundException,
+          InvalidArgumentException {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "Entry to deleteTaskComment(taskId= {}, taskCommentId= {})", taskId, taskCommentId);
+    }
+
+    taskService.deleteTaskComment(taskId, taskCommentId);
+
+    ResponseEntity<TaskCommentResource> result = ResponseEntity.noContent().build();
+
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug("Exit from deleteTaskComment(), returning {}", result);
+    }
+
+    return result;
+  }
+
+  @PutMapping(path = Mapping.URL_TASK_COMMENT)
+  @Transactional(readOnly = true, rollbackFor = Exception.class)
+  public ResponseEntity<TaskCommentResource> updateTaskComment(
+      @PathVariable String taskId,
+      @PathVariable String taskCommentId,
+      @RequestBody TaskCommentResource taskCommentResource)
+      throws NotAuthorizedException, TaskNotFoundException, TaskCommentNotFoundException,
+          InvalidArgumentException, ConcurrencyException {
+    if (LOGGER.isDebugEnabled()) {
+      LOGGER.debug(
+          "Entry to updateTaskComment(taskId= {}, taskCommentId= {}, taskCommentResource= {})",
+          taskId,
+          taskCommentId,
+          taskCommentResource);
+    }
+
+    ResponseEntity<TaskCommentResource> result;
+
+    if ((taskCommentId.equals(taskCommentResource.getTaskCommentId())
+        && (taskId.equals(taskCommentResource.getTaskId())))) {
+
+      TaskComment taskComment = taskCommentResourceAssembler.toModel(taskCommentResource);
+
+      taskComment = taskService.updateTaskComment(taskComment);
+      result = ResponseEntity.ok(taskCommentResourceAssembler.toResource(taskComment));
+    } else {
+      throw new InvalidArgumentException(
+          String.format(
+              "TaskCommentId ('%s') or TaskId ('%s') are not identical with the ids"
+                  + " of object in the payload which should be updated",
+              taskCommentId, taskCommentResource.getTaskId()));
```

Comment:
the second parameter should be a pathvariable taskId, right?

Reply:
Indeed